betaflight/betaflight
# Rename 'display_name' to 'pilot_name'; rename 'name' to 'craft_name' ;Add the 'MSP2_GET_TEXT' and 'MSP2_SET_TEXT' MSP commands

Fixes https://github.com/betaflight/betaflight-configurator/issues/1877

Configurator changes @ https://github.com/betaflight/betaflight-configurator/pull/2798

### scope

This PR renames:
  - the  `display_name` CLI/firmware property (i.e. `pilotConfigMutable()->displayName`) to  `pilot_name` (thus `pilotConfigMutable()->pilotName`). See https://github.com/betaflight/betaflight-configurator/pull/2798#issuecomment-1279710612 for the reasoning behind this change. 
  - the  `name` CLI/firmware property (i.e. `pilotConfigMutable()->name`) to  `craft_name` (thus `pilotConfigMutable()->craftName`).

---

This PR also adds two new MSP v2 commands (used to get/set the new `display_name` pilot profile config prop):

- `MSP2_GET_TEXT`
- `MSP2_SET_TEXT`

Both methods are used in the updated configurator and are meant for _generic_ text property retrieval/modification; Nevertheless, currently only the new `MSP2TEXT_PILOT_NAME` (`pilotConfigMutable()->pilotName`) and `MSP2TEXT_CRAFT_NAME` (`pilotConfigMutable()->craftName`) properties are supported.
  - See https://github.com/betaflight/betaflight/pull/11391#discussion_r855697952 for the initial suggestion
  - The new `MSP2TEXT_*` mappings (which are _not_ high-level MSP commands!) are defined in `src/main/msp/msp_protocol_v2_betaflight.h` 

### end goal

The end goal is to enable setting the `pilot name` property (previously known as `display name`) through the configurator UI. Its value can be show in the OSD (independently of the current `craft name` which can already be managed through the configurator UI). 
The `name` to `craft name` rename was done as a minor extra cleanup with no functional impact. 

### OSD backwards compatibility

The new `PILOT_NAME` OSD element (configured via `osd_pilot_name_pos`) has been added _in the same placeplace_ as the old `DISPLAY_NAME` (and the OSD element order is critical towards their configurator<>firmware mapping). Nevertheless this should not be a breaking change to the OSD (please correct me if I am wrong :pray:) as the updated configurator would show the correct field in that position depending on the MSP version. Also both `PILOT_NAME` and `DISPLAY_NAME` are semantically identical.

The `CRAFT_NAME` OSD element remains unaffected (as it was already properly named all-around).

### testing

Test firmware:

1. Get the updated configurator from https://github.com/betaflight/betaflight-configurator/pull/2798
2. See [the custom build installation video](https://youtu.be/I1uN9CN30gw) 
3. Download and flash your preferred target from the latest build below 
4. ~~Make sure that the MSP API version has been updated. Run `version` in the CLI and make sure that the `API` version is `1.45`. E.g:~~
    ```
    # version
    # Betaflight / MATEKF411RX (M41R) 4.3.0 Feb  7 2022 / 02:05:27 (811f0ca70) MSP API: 1.45
    ```
